src/go/MONIT/alert.go

```diff
@@ -0,0 +1,658 @@
+				break
+			case 2:
+				dif += strconv.Itoa(timeList[ind]) + "D "
+				break
+			case 3:
+				dif += strconv.Itoa(timeList[ind]) + "h "
+				break
+			case 4:
+				dif += strconv.Itoa(timeList[ind]) + "m "
+				break
+			case 5:
+				dif += strconv.Itoa(timeList[ind]) + "s "
+				break
+			default:
+				break
+			}
+		}
+	}
+
+	return
+}
+
+//Function for time difference between two time.Time objects
+func timeDiff(t1 time.Time, t2 time.Time, duration int) string {
+	if t1.After(t2) {
+		timeList := diff(t1, t2)
+		return timeDiffHelper(timeList) + "AGO"
+	}
+
+	timeList := diff(t2, t1)
+	if duration == 1 {
+		return timeDiffHelper(timeList)
+	}
+	return "IN " + timeDiffHelper(timeList)
+
+}
+
+//Helper function for Filtering
+func filterHelper(each alertData) int {
+
+	if service == "" && severity == "" && tag == "" {
+		return 1
+	} else if service == "" && severity == "" && tag == each.Tag {
+		return 1
+	} else if service == "" && severity == each.Severity && tag == "" {
+		return 1
+	} else if service == "" && severity == each.Severity && tag == each.Tag {
+		return 1
+	} else if service == each.Service && severity == "" && tag == "" {
+		return 1
+	} else if service == each.Service && severity == "" && tag == each.Tag {
+		return 1
+	} else if service == each.Service && severity == each.Severity && tag == "" {
+		return 1
+	} else if service == each.Service && severity == each.Severity && tag == each.Tag {
+		return 1
+	} else {
+		return 0
+	}
+}
+
+//Function for Filtering
+func filter() {
+
+	filteredAlerts = make(map[string]int)
+	for _, each := range allAlertData {
+		if filterHelper(each) == 0 {
+			continue
+		}
+		filteredAlerts[each.Name] = 1
+	}
+}
+
+//Sorting Logic
+type durationSorter []alertData
+
+func (d durationSorter) Len() int      { return len(d) }
+func (d durationSorter) Swap(i, j int) { d[i], d[j] = d[j], d[i] }
+func (d durationSorter) Less(i, j int) bool {
+	return d[i].EndsAt.Sub(d[i].StartsAt) <= d[j].EndsAt.Sub(d[j].StartsAt)
+}
+
+type severitySorter []alertData
+
+func (s severitySorter) Len() int      { return len(s) }
+func (s severitySorter) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s severitySorter) Less(i, j int) bool {
+	return severityLevel[s[i].Severity] < severityLevel[s[j].Severity]
+}
+
+type startAtSorter []alertData
+
+func (s startAtSorter) Len() int      { return len(s) }
+func (s startAtSorter) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s startAtSorter) Less(i, j int) bool {
+	return s[j].StartsAt.After(s[i].StartsAt)
+}
+
+type endsAtSorter []alertData
+
+func (e endsAtSorter) Len() int      { return len(e) }
+func (e endsAtSorter) Swap(i, j int) { e[i], e[j] = e[j], e[i] }
+func (e endsAtSorter) Less(i, j int) bool {
+	return e[j].EndsAt.After(e[i].EndsAt)
+}
+
+//Function for sorting alerts based on a passed label
+func sortAlert() {
+	severityLevel = make(map[string]int)
+	//TODO : Have to add all possibilities of severity levels.
+	severityLevel["info"] = 0
```

Comment:
Should be replaced with configuration provided by user and if no such configuration file is provided you may use some defaults.

Reply:
The configuration file feature has been added.
Few points:-
1) The configuration file is in JSON format:- 
```
{           
   "severity" : {
            "info" : 0,
            "warning" : 1,
            "medium" : 2,
            "high" : 3,
            "critical" : 4,
            "urgent" : 5
        }
}
```
2) -sconfig flag can be used to provide the filepath for the config file. 
`alert -sconfig="<parent-directory>/config.json"`
3) If -sconfig flag is not used, it will fetch "SEVERITY_CONFIG" environment variable which stores config file path.
4) If SEVERITY_CONFIG env variable is also not set then severity levels will have defaults.
```
sLevel.SeverityLevels["info"] = 0
sLevel.SeverityLevels["warning"] = 1
sLevel.SeverityLevels["medium"] = 2
```